cvat/apps/dataset_manager/task.py

```diff
@@ -70,9 +70,18 @@ def _merge_table_rows(rows, keys_for_merge, field_id):
     return list(merged_rows.values())
 
 class JobAnnotation:
-    def __init__(self, pk):
-        self.db_job = models.Job.objects.select_related('segment__task') \
-            .select_for_update().get(id=pk)
+    def __init__(self, pk, is_prefetched=False):
```

Comment:
This new argument is unclear to me - when and why it should be used? I don't see how the fact the data is (was?) prefetched affects this class.

Reply:
It also affects the sorting of images in the annotation file in cvat format. (We can get sorted sequence from db or sort in code, but I've chosen this approach because we use it for tasks)

Reply:
I don't see how this answers the questions above.


> We can get sorted sequence from db or sort in code, but I've chosen this approach because we use it for tasks

I see that sorting is performed in the newly added code when frames are fetched from DB. The DB itself doesn't impose any sorting.